runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/script/DefaultScriptEvaluator.java

```diff
@@ -114,21 +114,34 @@ public String getLanguage(String languageOrExtension) {
         return null;
     }
 
+    private org.graalvm.polyglot.Context.Builder getGraalEngineContextBuilder(Map<String,Object> variables) {
+      HostAccess access = HostAccess.newBuilder(HostAccess.ALL)
+        .targetTypeMapping(Value.class, Object.class, Value::hasArrayElements, v -> new LinkedList<>(v.as(List.class))).build();
+      org.graalvm.polyglot.Context.Builder ctx = org.graalvm.polyglot.Context.newBuilder("js")
+        .allowHostAccess(access);
+      for (String key : variables.keySet()) {
+        switch (key) {
+          case "ecmascript-version":
+            ctx.option("js."+key, variables.get(key).toString());
+            break;
+          default:
+            throw new RuntimeException("unsupported engine option: " + key);
```

Comment:
This should just ignore the invalid options rather than hijacking the entirety of input variables. E.g. this is now broken but should work
```yaml
- script: js
  in:
    myString: 'hello world'
  body: |
    print(myString);
````
results in
```text
Error: unsupported engine option: myString
```

Reply:
...apparently we need a test to make sure this isn't accidentally broken

Reply:
removal of the throw: 97df51573 

test: yes, i'd love to write a test or two! if it's not too much of a burden, can one of you all send me a _seed_ or ideas here? given that this is firing up a _VM_, i suspect there will be some strong opinions on how such a test is setup. is it a unit test with graal mocked, is it an integration test where we actually execute graal?

i'm happy to work on the cases, but seems like some teamwork may be warranted for the initial setup?

Reply:
depending on what needs to be tested, we have unit tests and integration tests that can handle most situations. Usual testing rules of thumb apply...all else equal, unit tests are preferable to integration tests. The ITs leverage [testcontainers](https://github.com/concord-workflow/testcontainers-concord) to test against a "real" concord instance.

> given that this is firing up a VM
For this specific instance, there isn't any extra VM being created. The default runtime covers the necessary dependencies.

There are existing tests similar to what needs to be done here. For ref:
 https://github.com/walmartlabs/concord/blob/68d57f52f412c94546815a5deaf8629bf9e9a3c5/runtime/v2/runner/src/test/java/com/walmartlabs/concord/runtime/v2/runner/MainTest.java#L437-L447

https://github.com/walmartlabs/concord/blob/68d57f52f412c94546815a5deaf8629bf9e9a3c5/runtime/v2/runner/src/test/resources/com/walmartlabs/concord/runtime/v2/runner/scriptVariablesSanitize/concord.yml#L1-L8

Add a similar test resource `<dir>/concord.yml` and a test in `MainTest`.

Reply:
ah, thanks! very helpful. i don't know how i missed those! a few tests sprinkled in:

3beb9d4ef

Reply:
realized I kept making JS errors and having formatting problems inline. moved that JS to a script file: 5b639ea13